### ISSUE TITLE
Issue with k8s.io/docs/concepts/workloads/controllers/job/

### ISSUE THREAD

**Issue by dorisyang-forever** [kubernetes/website#34433]
Env: windows + minikube

Try the first job example "pi"
job.batch/pi created

but when using: kubectl get pods    weird things happened:
**there are 5 pods created but ended in error, the status of each changing from ContainerCreating->running->error**
Finally all 5 pods error:
```
pi-77wsz   0/1     Error    0          31s
pi-9vt9k   0/1     Error    0          39s
pi-cg5lq   0/1     Error    0          9s
pi-vm685   0/1     Error    0          23s
pi-wfpwq   0/1     Error    0          16s
```
```
kubectl describe pods pi-wfpwq
```
```
Name:         pi-wfpwq
Namespace:    default
Priority:     0
Node:         minikube/192.168.49.2
Start Time:   Mon, 20 Jun 2022 23:36:08 -0500
Labels:       controller-uid=5ca39977-f3b1-4c69-964d-ef8b7cd2c50d
              job-name=pi
Annotations:  <none>
Status:       Failed
IP:           172.17.0.2
IPs:
  IP:           172.17.0.2
Controlled By:  Job/pi
Containers:
  pi:
    Container ID:  docker://f1128df21ccf4e2c21ed25f6ee12eb5493a3e3b4ba54a1c2908e9bed758f58ec
    Image:         perl
    Image ID:      docker-pullable://perl@sha256:d7fd57000721562918b96a687a216eddceb450436de4b89c2960fd7c5f2e34a7
    Port:          <none>
    Host Port:     <none>
    Command:
      perl
      -Mbignum=bpi
      -wle
      print bpi(2000)
    State:          Terminated
      Reason:       Error
      Exit Code:    255
      Started:      Mon, 20 Jun 2022 23:36:10 -0500
      Finished:     Mon, 20 Jun 2022 23:36:14 -0500
    Ready:          False
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-t8k5v (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  kube-api-access-t8k5v:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age    From               Message
  ----    ------     ----   ----               -------
  Normal  Scheduled  2m57s  default-scheduler  Successfully assigned default/pi-wfpwq to minikube
  Normal  Pulling    2m58s  kubelet            Pulling image "perl"
  Normal  Pulled     2m56s  kubelet            Successfully pulled image "perl" in 1.045504951s
  Normal  Created    2m56s  kubelet            Created container pi
  Normal  Started    2m56s  kubelet            Started container pi
```
The error code did not tell details:
```
    State:          Terminated
      Reason:       Error
      Exit Code:    255
```

**Comment by Shubham82** [comment]
Hi @dorisyang-forever 
It is a Duplicate of #34073,  There is a PR #34414 open for it.

I also tried with `image: perl:5.34.0` it's working fine.

**Comment by dorisyang-forever** [comment]
> Hi @dorisyang-forever It is a Duplicate of #34073, There is a PR #34414 open for it.
> 
> I also tried with `image: perl:5.34.0` it's working fine.

Thank you so much. 

Changed the yaml file, then it works! (Although the pods stay in "ContainerCreating" status for a long time)

apiVersion: batch/v1
kind: Job
metadata:
  name: pi
spec:
  template:
    spec:
      containers:
      - name: pi
        image: perl:5.34.0
        command: ["perl",  "-Mbignum=bpi", "-wle", "print bpi(2000)"]
      restartPolicy: Never
  backoffLimit: 4

**Comment by dorisyang-forever** [comment]
But after a while, the pod create is in **terminated** state, is it normal?
 kubectl get pods
NAME       READY   STATUS      RESTARTS   AGE
pi-xc89m   0/1     Completed   0          24s

Events:
  Type     Reason       Age                From               Message
  ----     ------       ----               ----               -------
  Normal   Scheduled    32s                default-scheduler  Successfully assigned default/pi-xc89m to minikube
  Normal   Pulled       31s                kubelet            Container image "perl:5.34.0" already present on machine
  Normal   Created      31s                kubelet            Created container pi
  Normal   Started      31s                kubelet            Started container pi
  **Warning  FailedMount  26s (x2 over 27s)  kubelet            MountVolume.SetUp failed for volume "kube-api-access-8ftx4" : object "default"/"kube-root-ca.crt" not registered**

**Comment by Shubham82** [comment]
It is not terminated in my case, i used minikube having k8s version v1.24.0

Which k8s version you are using.
Please see this issue #105204, this issue was in v1.22 and v1.23 and resolved in v1.22.9 and v1.23.6

**Comment by dorisyang-forever** [comment]
I am using k8s 1.24.0

**Comment by sftim** [comment]
This needs a fix that is similar to https://github.com/kubernetes/website/pull/34475

/language en
/kind bug
/triage accepted

**Comment by jihoon-seo** [comment]
Resolved by #34414.
/close
